File: modules/cubex-config/src/main/java/org/cubexmc/config/ResourceFiles.java
package org.cubexmc.config;

import java.io.File;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.List;
import org.cubexmc.core.CubexPlugin;

public final class ResourceFiles {

    private final CubexPlugin plugin;

    public ResourceFiles(CubexPlugin plugin) {
        this.plugin = plugin;
    }

    public boolean saveIfMissing(String resourcePath) {
        if (resourcePath == null || resourcePath.isBlank()) {
            return false;
        }
        File target = dataFile(resourcePath);
        if (target.exists()) {
            return false;
        }
        if ("config.yml".equals(resourcePath)) {
            plugin.saveDefaultConfig();
        } else {
            plugin.saveResource(resourcePath, false);
        }
        return true;
    }

    public List<String> saveIfMissing(Collection<String> resourcePaths) {
        if (resourcePaths == null || resourcePaths.isEmpty()) {
            return Collections.emptyList();
        }
        List<String> saved = new ArrayList<>();
        for (String resourcePath : resourcePaths) {
            if (saveIfMissing(resourcePath)) {
                saved.add(resourcePath);
            }
        }
        return saved;
    }

    public File dataFile(String resourcePath) {
        return new File(plugin.getDataFolder(), resourcePath);
    }

    public boolean exists(String resourcePath) {
        return dataFile(resourcePath).exists();
    }
}

File: modules/cubex-config/src/main/kotlin/org/cubexmc/config/ResourceFiles.kt
package org.cubexmc.config

import org.cubexmc.core.CubexPlugin
import java.io.File

class ResourceFiles(private val plugin: CubexPlugin) {
    fun saveIfMissing(resourcePath: String?): Boolean {
        if (resourcePath.isNullOrBlank()) return false
        if (dataFile(resourcePath).exists()) return false
        if (resourcePath == "config.yml") plugin.saveDefaultConfig() else plugin.saveResource(resourcePath, false)
        return true
    }

    fun saveIfMissing(resourcePaths: Collection<String>?): List<String> =
        resourcePaths?.filter { saveIfMissing(it) } ?: emptyList()

    fun dataFile(resourcePath: String): File = File(plugin.dataFolder, resourcePath)

    fun exists(resourcePath: String): Boolean = dataFile(resourcePath).exists()
}
